app/assets/javascripts/discourse/app/controllers/discovery/topics.js

```diff
@@ -24,10 +24,17 @@ const controllerOpts = {
   showTopicPostBadges: not("new"),
   redirectedReason: alias("currentUser.redirected_to_top.reason"),
 
-  @discourseComputed("model.filter", "site.show_welcome_topic_banner")
-  showEditWelcomeTopicBanner(filter, showWelcomeTopicBanner) {
+  @discourseComputed(
+    "model.filter",
+    "site.show_welcome_topic_banner",
+    "model.listParams.f"
+  )
+  showEditWelcomeTopicBanner(filter, showWelcomeTopicBanner, hasListParams) {
     return (
-      this.currentUser?.staff && filter === "latest" && showWelcomeTopicBanner
+      this.currentUser?.staff &&
+      filter === "latest" &&
```

Comment:
If the homepage is `/categories`, should this still show on `/latest`? 

I think we need to improve the acceptance tests on `welcome-topic-banner-test.js` to account for a few of these cases (`f` query param, non-latest routes, different homepages).

Reply:
> If the homepage is `/categories`, should this still show on `/latest`?

Yes I think so, this should only show on the regular topic-list regardless of what is set as the homepage.




> I think we need to improve the acceptance tests on `welcome-topic-banner-test.js` to account for a few of these cases (`f` query param, non-latest routes, different homepages).

👍

Reply:
> I think we need to improve the acceptance tests on `welcome-topic-banner-test.js`

How about these test additions?